apps/ewallet_db/lib/ewallet_db/exchange_pair.ex

```diff
@@ -0,0 +1,171 @@
+defmodule EWalletDB.ExchangePair do
+  @moduledoc """
+  Ecto Schema representing an exchange pair.
+
+  # What is an exchange rate?
+
+  The exchange rate is the amount of the destination token (`to_token`) that will be received
+  when exchanged with one unit of the source token (`from_token`).
+
+  For example:
+
+  ```
+  %EWalletDB.ExchangePair{
+    name: "AAA/BBB",
+    from_token: AAA,
+    to_token: BBB,
+    rate: 2.00,
+    reversible: true
+  }
+  ```
+
+  The struct above means that 1 AAA can be exchanged for 2 AAA.
+  Also, because of `reversible: true`, 2 BBB can be exchanged for 1 AAA.
+  """
+  use Ecto.Schema
+  use EWalletDB.SoftDelete
+  use EWalletDB.Types.ExternalID
+  import Ecto.Changeset
+  import EWalletDB.Helpers.Preloader
+  alias Ecto.UUID
+  alias EWalletDB.{Repo, Token}
+
+  @primary_key {:uuid, UUID, autogenerate: true}
+
+  schema "exchange_pair" do
+    external_id(prefix: "exg_")
+
+    field(:name, :string)
+    field(:rate, :float)
+    field(:reversible, :boolean)
```

Comment:
A little bit torn about handling `reversible`, can we not have an explicit reverse from/to pair, as in `from: "AAA", to: "BBB"` and `from: "BBB", to: "AAA"` as a separate exchange pairs? I think it might be simpler, logic-wise. Thoughts @T-Dnzt?

Reply:
I had reversible because I found that a few libraries in other languages do support reversible rates. But I'm ok to drop reversible support. It helps to reduce overall complexity like sirn said. Final decision for you @T-Dnzt.

Reply:
Let's make it explicit by creating a separate exchange pair then. But can we keep the reversible parameter in the endpoints and automatically create that second separate pair from the API? Makes things faster than having to create 2 pairs every time :)

Reply:
Oh, if we do this then there is no need to add the `reversible` field to the JSON

Reply:
@T-Dnzt If we go with auto-create the opposite pair,`reversible` sounds like an attribute that gets attached to the direct pair.

I propose renaming it to `create_opposite_pair: true/false` or something similar?

Reply:
Sure! Just has to be in the API anyway, probably not in this PR? 🤔

Reply:
Yes can take it as a separate PR. But to make sure this PR is compatible with the direction above, I will remove reversible from `ExchangePair.cast()` and default the value to `false`.

Reply:
Ok so I decided to remove the `reversible` field from the database instead. So we don't need to add another DB migration (along with index changes) later.